doc/MiniLanguageTutorial.md

```diff
@@ -480,37 +563,278 @@ Mini never automatically converts types to make an operation succeed.  Programme
+The default values for each type are as follows:
+
+* `uint`, `int`, `bytes32`, `address`:  
+  `0`, represented in `AVM` as an `Int`.
+* `bool`:  
+  `false`, this is also a `0` `Int` in the underlying AVM.
+* `buffer`:  
+  A buffer with length `0`, attempting to read any bytes past the end of a buffer will be treated as reading `0` values.
+* `any`:  
+  A length `0` tuple, `()`.
+* `union`:  
+  The default value of a union type is the default value of its leftmost variant.
+* `tuple`:  
+  The default value of a tuple is a wrapping tuple, with leaves equal to the default values of each field type respectively. 
+* `struct`:  
+  The default value of a struct is a wrapping tuple, with leaves equal to the default values of each field type respectively.
+* `option`:  
+  The default value of all `option` types is the `None` variant, in AVM this is represented as a length 1 tuple containing an `Int` value `0`
+* `func`:  
+  The default value of all `func` types is a `CodePoint` 
+* `map`:  
+  An empty map, in AVM this is represented as a length 2 `Tuple` with both fields as an `Int` with value `0`.
+* `array`:  
+  An `array` of *type* has a default value of an array with length 1 containing the default value of *type*.
+  In AVM this is represented by a length 3 `Tuple` containing, `Int`s with a value of `0` for the first two fields, 
+  and a length 8 `Tuple` containing 
+* `fixed array`:  
+  A default value for a `fixed array` of type *type* and length *length* is a fixed array of length *length*, where every array element has a value of the default value of *type*.
+  See the AVM representation section for more information of how the memory for this value is layed out.
+* `nominal`, `specialized generic`:  
+  The default value for a nominal or specialized generic type is the default value of its representation.
+
+While non-specialized generic types should never be created, if they are they will have a default value of `()`.
+
+## AVM Representation
+
+Types in mini are internally represented as AVM values.
+
+1. uint:  
+   Always an `Int`, any `Int` value is allowable for this type.
+1. int:  
+   Always an `Int`, any `Int` value is allowable for this type.
+1. bool:  
+   Always an `Int`, the values `0` and `1` are allowed for this type, but nothing else.
+1. buffer:  
+   Always a `Buffer`, any value is allowable for this type.
+1. bytes32:  
+   Always an `Int`, any value is allowable for this type.
+1. address:  
+   Always an `Int`, only values up to `2^160-1` are allowed for this type, where the `Int` must be interpreted as unsigned.
+1. void:  
+   Represents lack of a value, whenever a value of type `void` is present, then it is assumed there is no value there.
+1. struct:  
+   Represented by a wrapping tuple containing values of the types of each of its fields, in order from top to bottom.
+   Field names have no impact on the allowable values for the type.
```

Comment:
Field names do affect the values, though, because they affect the type.  What is accurate, I think, is that field names have no affect on the AVM representation of the type.

Reply:
By values I meant AVM values, I've clarified.